### PR TITLE
Introduce shared build methods

### PR DESCRIPTION
**Pull request by pavelhoral**
This PR introduces new Node module `@wrensecurity/commons-ui-build` that contains shared key gulp tasks.

I have to admit that this brings the project closer to its previous state with Grunt. However it is still a good step towards potentially more modern build tooling in the future. This change also allows us to perform any upgrade in a more controllable fashion.

New module is supposed to be distributed as Maven artifact (we are not switching to NPM yet). Downstream projects will include it as a local dependency provided by unpack task of maven-dependency-plugin.

### CODE REVIEW

**Comment by pavelhoral** [comment]
Hm... for some reason the build fails in GH Actions, but works completely fine on my computer. Will need to investigate it a bit. However the PR can be reviewed.

File: packages/build/scripts/build.mjs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    input: "src/index.mjs",
+    plugins: [nodeResolve()],
+    external: /node_modules/
+});
+await bundle.write({
+    format: "cjs",
+    file: join(TARGET_PATH, "index.cjs")
+});
+
+await mkdir(TARGET_PATH, { recursive: true });
```
Comment (karelmaxa) [inline packages/build/scripts/build.mjs:36]: Target directory has already been created (see line 24).

File: packages/build/src/index.mjs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+/**
+ * @type {Object} ModuleResourcesOptions
+ * @property {string} dest - target path (defaults to `dist`)
+ * @property {string|URL} path - base path for the NodeJS module resolver
+ */
+
+/**
+ * Copy specific Node module resources.
+ * @param {Record<string, string>} resources - resource map
+ * @param {useModuleResources} [options]
```
Comment (karelmaxa) [inline packages/build/src/index.mjs:36]: The correct parameter type should be `ModuleResourcesOptions`.

**Review by karelmaxa — CHANGES_REQUESTED** [review]

**Review by karelmaxa — APPROVED** [review]
LGTM.

**Pull request merged**